src/compas_fab/robots/wrench.py

```diff
@@ -0,0 +1,353 @@
+		http://docs.ros.org/kinetic/api/geometry_msgs/html/msg/WrenchStamped.html
+
+		Parameters
+		----------
+		data : :obj:`dict`
+			The data dictionary.
+
+		Returns
+		-------
+		Wrench
+			The constructed wrench.
+		
+		Note:
+			Used from creating Wrench from ROS message
+		"""
+		wrench = [
+			msg["wrench"]["force"]["x"],
+			msg["wrench"]["force"]["y"],
+			msg["wrench"]["force"]["z"],
+			msg["wrench"]["torque"]["x"],
+			msg["wrench"]["torque"]["y"],
+			msg["wrench"]["torque"]["z"]
+		]
+		return cls(wrench, bias = bias, mass = mass, center_of_mass = center_of_mass)
+	
+	@classmethod
+	def by_samples(cls, wrenches, bias = None, mass = None, center_of_mass = None, proportiontocut = 0.0):
+		"""
+		Construct the compensator by sampled data, allowing to filter lists of wrenches
+		"""
+		wrench = stats.trim_mean(wrenches, proportiontocut, axis=0).tolist()
+		return cls(wrench, bias = bias, mass = mass, center_of_mass = center_of_mass)
+	
+	# ==========================================================================
+	# descriptors
+	# ==========================================================================
+
+	@property
+	def data(self):
+		"""Returns the data dictionary that represents the wrench.
+
+		Returns
+		-------
+		dict
+			The wrench data.
+
+		"""
+		return {'wrench': {
+			'force': {
+				'x': self.wrench[0],
+				'y': self.wrench[1],
+				'z': self.wrench[2],
+			},
+			'torque': {
+				'x': self.wrench[3],
+				'y': self.wrench[4],
+				'z': self.wrench[5]}}}
+
+	def to_data(self):
+		"""Returns the data dictionary that represents the wrench.
+
+		Returns
+		-------
+		dict
+			The wrench data.
+		"""
+		return self.data
+	
+	def to_list(self):
+		"""Returns the list that represents the wrench.
+
+		Returns
+		-------
+		list
+			The wrench list.
+		"""
+		return self.wrench
+	
+	def to_lists(self):
+		"""Returns the 2 lists that represents the forces and torques.
+
+		Returns
+		-------
+		lists
+			The forces list.
+			The torques list.
+		"""
+		return list(self.wrench[0:3]), list(self.wrench[3:6])
+
+	# ==========================================================================
+	# representation
+	# ==========================================================================
+
+	def __repr__(self):
+		return '''Wrench(Fx {0:.2f}, Fy {1:.2f}, Fz {2:.2f}, Tx {3:.2f}, Ty {4:.2f}, Tz {5:.2f})'''.format(*self.wrench)
+
+	# ==========================================================================
+	# helpers
+	# ==========================================================================
+
+	def copy(self):
+		"""Make a copy of this ``Wrench``.
+
+		Returns
+		-------
+		Wrench
+			The copy.
+		"""
+		cls = type(self)
+		return cls(copy.copy(self.wrench))
+
+	# ==========================================================================
+    # methods
+    # ==========================================================================
+	
+	@staticmethod
+	def binary(wrench, forces_threshold = 10.0, torques_threshold = 5.0):
```

Comment:
not sure where this is needed, more explanation?, example?

Reply:
I discarded this, it was too project specific